nameserver/server.go

```diff
@@ -194,17 +206,6 @@ func (s *DNSServer) Start() error {
 		Info.Printf("[dns] Cache: %d entries", s.cache.Capacity())
 	}
 
-	pc, err := net.ListenPacket("udp", s.ListenAddr)
-	if err != nil {
-		return err
-	}
-	s.pc = pc
-
-	_, port, err := net.SplitHostPort(pc.LocalAddr().String())
-	if err != nil {
-		return err
-	}
-	s.ListenAddr = fmt.Sprintf(":%s", port)
```

Comment:
Well, I think that `New*()` methods should not try to be too smart. For example, imagine you create a `NewMyDatabaseClient()`: I don't think it should create a connection pool until you `Start()` or `Connect()` or something else... It is a pattern I always try to follow, but this is my personal opinion... 

And, yes, I've thought several times about refactoring some of our components from `Start()`/`Stop()` to  the `ListenAndServe()`/`Shutdown()` _standard_ pattern. That would make things more clear, a much better API, but I've never done it because it would also break the convenient [SignalReceiver interface](https://github.com/weaveworks/weave/blob/master/common/signals.go#L11) found in `common/signal.go`...